FTP Front-end Project/src/components/ModalWithForm/ModalWithForm.jsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import { useEffect, useRef, useState } from "react";
+import "./ModalWithForm.css";
+
+const ModalWithForm = ({
+  title,
+  onClose,
+  children,
+  buttonText,
+  buttonText2,
+  isOpen,
+  onSubmit,
+  onSign,
+}) => {
+  const modalRef = useRef(null);
+
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (modalRef.current && !modalRef.current.contains(event.target)) {
+        onClose();
+      }
+    };
+
+    if (isOpen) {
+      document.addEventListener("mousedown", handleClickOutside);
+    } else {
+      document.removeEventListener("mousedown", handleClickOutside);
+    }
+
+    return () => document.removeEventListener("mousedown", handleClickOutside);
+  }, [isOpen, onClose]);
+
+  return (
+    <div className={`modal ${isOpen ? "active" : ""} modal__type_${title}`}>
```

Comment:
`active`


You should use a **modifier** or `modal` for opening and closing: `modal_active`, according to BEM